metpy/calc/thermo.py

```diff
@@ -1038,3 +1038,99 @@ def most_unstable_parcel(p, temperature, dewpoint, heights=None,
     theta_e = equivalent_potential_temperature(p_layer, T_layer)
     max_idx = np.argmax(theta_e)
     return p_layer[max_idx], T_layer[max_idx], Td_layer[max_idx]
+
+
+@exporter.export
+@check_units('[pressure]', '[temperature]', '[temperature]')
+def mixed_parcel(p, temperature, dewpoint, parcel_start_pressure=None, bottom=None,
+                 heights=None, depth=100 * units.hPa, interpolate=True):
+    r"""Calculate the properties of a parcel mixed from a layer.
+
+    Determines the properties of an air parcel that is the result of complete mixing of a
+    given atmospheric layer.
+
+    Parameters
+    ----------
+    p : array-like
```

Comment:
should these be `pint.Quantity` rather than array-like since we're checking units?